actor/builtin/miner/miner.go

```diff
@@ -461,6 +472,34 @@ func (ma *Actor) IsBootstrapMiner(ctx exec.VMContext) (bool, uint8, error) {
 	return ma.Bootstrap, 0, nil
 }
 
+// GetPoStState returns whether the miner's last submitPoSt within the proving period,
+// late or after the generation attack threshold.
+func (ma *Actor) GetPoStState(ctx exec.VMContext) (*big.Int, uint8, error) {
+
+	var state State
+	bh := ctx.BlockHeight()
+	out, err := actor.WithState(ctx, &state, func() (interface{}, error) {
+		generationAttackGracePeriod := GenerationAttackTime(state.SectorSize)
+		// Don't check lateness unless there is storage to prove
+		if len(state.ProvingSet.Values()) == 0 {
```

Comment:
There's a fancy `ProvingSet.Size()` function you can use now

Reply:
I wasn't sure about it, now I know!